WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -110,14 +143,54 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     func cardPaymentTapped() {
         Task { @MainActor in
             showsCreatingOrderSheet = true
-            let order = try await createTestOrder()
+            let finalOrder = orderService.orderFrom(posOrder: order)
             showsCreatingOrderSheet = false
-            let _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
+            let result = try await cardPresentPaymentService.collectPayment(for: finalOrder, using: .bluetooth)
 
             // TODO: Here we should present something to show the payment was successful or not,
             // and then clear the screen ready for the next transaction.
+            switch result {
+            case .success(let cardPresentPaymentTransaction):
+                print("🟢 [POS] Payment successful: \(cardPresentPaymentTransaction)")
+            case .cancellation:
+                print("🟡 [POS] Payment cancelled")
+            }
         }
     }
+
+    func updateOrderStatus(_ status: OrderStatusEnum) {
+        guard let order else {
+            return
+        }
+        Task { @MainActor
+            [weak self] in
+            guard let self else {
+                throw OrderSyncError.selfDeallocated
+            }
+            do {
+                let finalizedOrder = try await self.orderService.updateOrderStatus(posOrder: order, status: status)
+                self.order = finalizedOrder
+                print("🟢 [POS] Synced order: \(finalizedOrder)")
+            } catch {
+                print("🔴 [POS] Error syncing order: \(error)")
+            }
+        }
+    }
+
+    func calculateAmountsTapped() {
+        createOrUpdateOrder()
+    }
+
+    private func createOrUpdateOrder() {
+        finalItems = itemsInCart
```

Comment:
I added this `finalItems` property to use it as a trigger for order calculation, since "normal" `itemsInCart` are updated on every add/remove and we need to calculate totals only on checkout.

What do you think if I change/remove `observeProductsInCartForRemoteOrderSyncing`, not to use Combine but have a "manual" async method that does the calculation and order sync?